src/inmanta/module.py

```diff
@@ -1489,6 +1514,16 @@ def requires(self) -> "List[Requirement]":
             reqs.append(reqe)
         return reqs
 
+    def get_all_requires(self) -> List[str]:
+        """
+        :return: all modules required by an import from any sub-modules, with all constraints applied
+        """
+        # get all constraints
+        spec: Dict[str, Optional["Requirement"]] = {req.project_name: req for req in self.requires()}
+        # find all imports
+        imports = {imp.name.split("::")[0] for subm in sorted(self.get_all_submodules()) for imp in self.get_imports(subm)}
+        return sorted([str(spec[r]) if spec.get(r) else r for r in imports])
```

Comment:
I find it a bit strange to have this method return constraint strings. Why not just return `List[Requirement]`? If we do return strings, I think the docstring needs to be updated to state the strings contain constraints as well as the module names.

Reply:
updated